vehicle/src/Vehicle/Compile/Type/Monad.hs

```diff
@@ -62,13 +63,14 @@ import Vehicle.Data.Variable.Bound.Context.Generic
 import Vehicle.Data.Variable.Free.Context
 
 runTypeCheckerTInitially ::
-  (Monad m) =>
+  (Monad m, Hashable builtin) =>
   FreeCtx builtin ->
   InstanceDatabase builtin ->
   TypeCheckerT builtin m a ->
   m a
-runTypeCheckerTInitially freeCtx instanceCandidates e =
-  fst <$> runTypeCheckerT freeCtx instanceCandidates emptyTypeCheckerState e
+runTypeCheckerTInitially freeCtx instanceDatabase e = do
+  let state = (\s -> s {instanceCandidates = instanceDatabase}) emptyTypeCheckerState
```

Comment:
Can write this as:
```haskell
emptyTypeCheckerState { instanceCandidates = instanceDatabase } 
```
In fact it would make more sense for the `emptyTypeCheckerState` function to take the instance database as a direct argument?